pkg/util/cfg/dynamic.go

```diff
@@ -22,14 +22,15 @@ func DynamicUnmarshal(dst DynamicCloneable, args []string, fs *flag.FlagSet) err
 		// Next populate the config from the config file, we do this to populate the `common`
 		// section of the config file by taking advantage of the code in ConfigFileLoader which will load
 		// and process the config file.
-		ConfigFileLoader(args, "config.file"),
+		ConfigFileLoader(args, "config.file", true),
 		// Apply any dynamic logic to set other defaults in the config. This function is called after parsing the
 		// config files so that values from a common, or shared, section can be used in
 		// the dynamic evaluation
 		dst.ApplyDynamicConfig(),
 		// Load configs from the config file a second time, this will supersede anything set by the common
 		// config with values specified in the config file.
-		ConfigFileLoader(args, "config.file"),
+		// Second marshaling is not strict because the config is already populated and that causes errors with maps.
+		ConfigFileLoader(args, "config.file", false),
```

Comment:
Can you please expand a bit on why this was necessary to add?
This is such a fundamental behaviour in Loki, and we should be careful about changing it

"causes errors with maps" - what did you see here?

Reply:
By loading the config file twice and unmarshaling it into the same object, using strict yaml unmarshal causes a `already set in map` error with the `Clients` config, because it's a map that already has the keys we are trying to unmarshal into it.

Reply:
Thanks for the explanation but I'd love it if we could figure out how to solve this without changing this.
I'm concerned making this unstrict will introduce some subtle bugs.
I don't know this part of the code well, which is contributing to my skepticism.

Reply:
It's not actually an issue since we already marshal the file with strict the first time, so any problems with the yaml would result in an error the first time we do the marshaling, so the second one we can confidently marshal without being strict

Reply:
I'm sure there was a reason for the strict parsing the second time around, though; @slim-bean I see you initially wrote this; could you weigh in pls?